src/core/server/graph/extensions/LoggerExtension.ts

```diff
@@ -33,9 +39,36 @@ export function logQuery(
         : // A persisted query was not provided, parse the operation metadata
           // out from the document.
           getOperationMetadata(document)),
-    },
-    "graphql query"
-  );
+    };
+
+    // CORL-933: We are checking we have a logger as sometimes
+    // we are seeing no logger on the graph context. We're not
+    // sure why this is the case, but we want to catch when this
+    // is happening and not throw an error.
+    if (!ctx || !ctx.logger) {
+      // Catch the context and log it out, we need to know what is
+      // going on here
+      logger.warn({ ctx }, "unable to find logger on GraphContext");
```

Comment:
Avoid logging the context directly (it _will_ try to serialize everything) and instead just do a lighter reference:

```suggestion
      logger.warn({ ctx: Boolean(ctx) }, "unable to find logger on GraphContext");
```